libs/application-generic/src/usecases/tier-restrictions-validator/tier-restrictions-validator.usecase.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@nestjs/common';
+
+import { ApiServiceLevelEnum, StepTypeEnum } from '@novu/shared';
+import { CommunityOrganizationRepository } from '@novu/dal';
+
+import { TierRestrictionsValidatorCommand } from './tier-restrictions-validator.command';
+import {
+  ErrorEnum,
+  TierRestrictionsValidatorResponse,
+} from './tier-restrictions-validator.response';
+import {
+  FREE_TIER_MAX_DELAY_DAYS,
+  BUSINESS_TIER_MAX_DELAY_DAYS,
+  MAX_DELAY_FREE_TIER,
+  MAX_DELAY_BUSINESS_TIER,
+} from './tier-restrictions-validator.consts';
+
+@Injectable()
+export class TierRestrictionsValidatorUsecase {
+  constructor(
+    private organizationRepository: CommunityOrganizationRepository,
+  ) {}
+
+  async execute(
+    command: TierRestrictionsValidatorCommand,
+  ): Promise<TierRestrictionsValidatorResponse> {
+    return this.validateControlValuesByTierLimits(
+      command.organizationId,
+      command.deferDurationMs,
+      command.stepType,
+    );
+  }
+
+  private async validateControlValuesByTierLimits(
+    organizationId: string,
+    deferDurationMs?: number,
+    stepType?: StepTypeEnum,
+  ): Promise<TierRestrictionsValidatorResponse> {
+    const controlValueNeedTierValidation =
+      stepType === StepTypeEnum.DIGEST || stepType === StepTypeEnum.DELAY;
+
+    if (!controlValueNeedTierValidation || !deferDurationMs) {
+      return null;
+    }
+
+    const issues: TierRestrictionsValidatorResponse = [];
+    const organization =
+      await this.organizationRepository.findById(organizationId);
+    const tier = organization?.apiServiceLevel;
+
+    if (
+      tier === undefined ||
+      tier === ApiServiceLevelEnum.BUSINESS ||
+      tier === ApiServiceLevelEnum.ENTERPRISE
+    ) {
+      if (deferDurationMs > MAX_DELAY_BUSINESS_TIER) {
+        issues.push({
+          error: ErrorEnum.TIER_LIMIT_EXCEEDED,
+          message:
+            `The maximum delay allowed is ${BUSINESS_TIER_MAX_DELAY_DAYS} days.` +
+            'Please contact our support team to discuss extending this limit for your use case.',
+        });
+      }
+    }
+
+    if (tier === ApiServiceLevelEnum.FREE) {
+      if (deferDurationMs > MAX_DELAY_FREE_TIER) {
+        issues.push({
+          error: ErrorEnum.TIER_LIMIT_EXCEEDED,
+          message:
+            `The maximum delay allowed is ${FREE_TIER_MAX_DELAY_DAYS} days.` +
+            'Please contact our support team to discuss extending this limit for your use case.',
+        });
+      }
```

Comment:
I would make something more structured here, the issue passed: true/ false and some metadata to build the msg, I wouldn't contain the msgs here.

Reply:
I feel that use the if statments to define the max values, than validate against the comuted value and return the pass / fail, max value , max unit would be a cleaner approach

Reply:
not sure i am following your thought here, can you elaborate?